code/modules/projectiles/guns/ballistic/launchers.dm

```diff
@@ -139,19 +139,33 @@
 /obj/item/gun/ballistic/rocketlauncher/update_icon()
 	icon_state = "[initial(icon_state)]-[chambered ? "1" : "0"]"
 
-/obj/item/gun/ballistic/rocketlauncher/suicide_act(mob/user)
+/obj/item/gun/ballistic/rocketlauncher/suicide_act(mob/living/user)
+	user.visible_message("<span class='warning'>[user] aims [src] at the ground! It looks like [user.p_theyre()] performing a sick rocket jump!<span>", \
+		"<span class='userdanger'>You aim [src] at the ground to perform a bisnasty rocket jump...</span>")
 	if(can_shoot())
-		handle_suicide(user, user, null, TRUE)
+		playsound(src, 'sound/vehicles/rocketlaunch.ogg', 80, 1, 5)
+		item_flags |= NODROP
+		user.Stun(75)
+		animate(user, pixel_z = 300, time = 30, easing = LINEAR_EASING)
+		sleep(70)
+		animate(user, pixel_z = 0, time = 5, easing = LINEAR_EASING)
+		sleep(5)
+		item_flags &= ~NODROP
```

Comment:
check to make sure it doesn't have this flag from the start, don't want to remove this if something else sets it.

Reply:
o tru

Reply:
I'm not a fan of doing this, I'd prefer the rocket launcher just dropped out of your hand, or delay their actual death until they return to the ground, instead of stunning them or whatever